packages/flutter_tools/lib/src/commands/test.dart

```diff
@@ -77,7 +79,12 @@ class TestCommand extends FlutterCommand {
         negatable: false,
         help: 'Whether matchesGoldenFile() calls within your test methods should\n'
               'update the golden files rather than test for an existing match.',
-      );
+      )
+      ..addOption('concurrency',
+        abbr: 'j',
+        defaultsTo: math.max<int>(1, platform.numberOfProcessors - 2).toString(),
```

Comment:
One CPU for the test driver, and one CPU for the user to actually use the computer. If we peg 100% of the CPU, then the machine will be unusable.

Reply:
In my experience, the machine isn't really unusable even when using all the cores, since users are usually just editing or browsing.  I have often run C++ compiles at -j32 on my machine without really slowing down my own editing, command line operations, etc. much at all, even when all 32 cores were busy.

Also, is the test driver really doing much other than waiting for tests to complete?

And for Cirrus, we don't have to account for a user anyhow, since it has the machine to itself.